In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("Naive-Bayes-Classification-Data.csv")

In [3]:
df

,glucose,bloodpressure,diabetes
0,40,85,0
1,40,92,0
2,45,63,1
3,45,80,0
4,40,73,1
...,...,...,...
990,45,87,0
991,40,83,0
992,40,83,0
993,40,60,1


In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 995 entries, 0 to 994
Data columns (total 3 columns):
 #   Column         Non-Null Count  Dtype
---  ------         --------------  -----
 0   glucose        995 non-null    int64
 1   bloodpressure  995 non-null    int64
 2   diabetes       995 non-null    int64
dtypes: int64(3)
memory usage: 23.4 KB


In [4]:
df.isna().sum()

glucose          0
bloodpressure    0
diabetes         0
dtype: int64

In [22]:
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
from sklearn.naive_bayes import GaussianNB
from sklearn.metrics import accuracy_score,classification_report,confusion_matrix

In [6]:
x = df.drop("diabetes",axis=1)
y = df["diabetes"]

In [7]:
x_train, x_test, y_train, y_test = train_test_split(
    x,y,test_size = 0.20, random_state = 42
)

In [8]:
scaler = StandardScaler()

In [9]:
x_train_scaled = scaler.fit_transform(x_train)
x_test_scaled = scaler.transform(x_test)

In [17]:
x_train_scaled

array([[ 0.10631451,  1.13952665],
       [ 0.10631451, -0.14556881],
       [-0.64927789, -0.14556881],
       ...,
       [-1.40487028, -0.14556881],
       [-0.64927789,  1.46080051],
       [ 0.10631451,  1.67498309]], shape=(796, 2))

In [23]:
#x_test_scaled

In [24]:
model = GaussianNB()

In [11]:
model

GaussianNB()

In [12]:
model.fit(x_train_scaled,y_train)

GaussianNB()

In [13]:
y_pred = model.predict(x_test_scaled)

In [14]:
y_pred

array([1, 1, 1, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 1, 1, 1, 1, 0, 0, 1,
       1, 0, 0, 0, 1, 1, 0, 0, 0, 1, 0, 1, 0, 1, 1, 0, 0, 0, 1, 0, 0, 0,
       0, 0, 1, 1, 1, 1, 0, 1, 0, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 0, 1,
       1, 0, 1, 1, 0, 0, 0, 0, 1, 1, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 1,
       0, 1, 0, 0, 1, 0, 1, 1, 0, 1, 1, 1, 1, 1, 1, 0, 0, 1, 1, 0, 1, 0,
       1, 0, 1, 0, 0, 1, 0, 0, 1, 1, 1, 0, 0, 0, 0, 1, 1, 0, 1, 1, 1, 0,
       0, 0, 1, 1, 1, 1, 0, 1, 1, 0, 1, 1, 1, 1, 0, 0, 1, 1, 0, 1, 0, 0,
       1, 1, 1, 0, 0, 1, 1, 1, 0, 1, 1, 1, 0, 0, 1, 1, 1, 0, 1, 0, 0, 1,
       0, 1, 1, 1, 1, 1, 0, 0, 0, 1, 1, 1, 0, 1, 1, 1, 1, 0, 0, 1, 0, 0,
       0])

In [15]:
comparison = pd.DataFrame({
    "Actual": y_test,
    "Prediction": y_pred
})

In [17]:
comparison.head(10)

,Actual,Prediction
920,1,1
525,1,1
567,1,1
657,0,0
633,0,0
429,0,0
857,1,1
712,1,0
174,0,0
604,1,1


In [19]:
accuracy = accuracy_score(y_test,y_pred)*100

print(f"accuracy: {accuracy:.2f}%")

accuracy: 92.96%


In [20]:
print("Classification Report :\n", classification_report(y_test,y_pred))

Classification Report :
               precision    recall  f1-score   support

           0       0.92      0.92      0.92        93
           1       0.93      0.93      0.93       106

    accuracy                           0.93       199
   macro avg       0.93      0.93      0.93       199
weighted avg       0.93      0.93      0.93       199



In [25]:
cm = confusion_matrix(y_test,y_pred)

In [30]:
print(cm)

[[86  7]
 [ 7 99]]


In [31]:
for i in range(45):
    x_train, x_test, y_train, y_test = train_test_split(
        x,y,test_size = 0.2,random_state = i
    )
    model = GaussianNB()
    model.fit(x_train,y_train)
    print(i, accuracy_score(y_test,model.predict(x_test)))

0 0.9095477386934674
1 0.9396984924623115
2 0.9396984924623115
3 0.9396984924623115
4 0.9296482412060302
5 0.9597989949748744
6 0.9346733668341709
7 0.9246231155778895
8 0.914572864321608
9 0.9396984924623115
10 0.9346733668341709
11 0.8994974874371859
12 0.9547738693467337
13 0.9346733668341709
14 0.9095477386934674
15 0.9246231155778895
16 0.9346733668341709
17 0.914572864321608
18 0.9246231155778895
19 0.914572864321608
20 0.9246231155778895
21 0.9346733668341709
22 0.9396984924623115
23 0.9246231155778895
24 0.949748743718593
25 0.9246231155778895
26 0.9045226130653267
27 0.9698492462311558
28 0.914572864321608
29 0.9547738693467337
30 0.949748743718593
31 0.9447236180904522
32 0.9597989949748744
33 0.9296482412060302
34 0.9195979899497487
35 0.914572864321608
36 0.9396984924623115
37 0.9246231155778895
38 0.9246231155778895
39 0.9195979899497487
40 0.949748743718593
41 0.9296482412060302
42 0.9296482412060302
43 0.949748743718593
44 0.949748743718593


In [32]:
from  sklearn.model_selection import cross_val_score

In [33]:
scores = cross_val_score(GaussianNB(),x,y,cv = 5)
print(scores)

[0.95979899 0.96482412 0.90954774 0.90452261 0.92462312]


In [34]:
print(scores.mean()*100)

93.26633165829146


## Model Evaluation

The Diabetes Classification model achieved an accuracy of **92.96%** on the test dataset.

The classification report shows that both classes performed well, with precision, recall, and F1-score around **92%–93%**.

The confusion matrix shows:
- **86** instances of Class 0 were correctly classified.
- **99** instances of Class 1 were correctly classified.
- **7** Class 0 instances were incorrectly classified as Class 1.
- **7** Class 1 instances were incorrectly classified as Class 0.

The model achieved an average **cross-validation score of 93.27%**, indicating good and consistent performance across different data splits.

### Conclusion

Overall, the model performed well on the Diabetes dataset, achieving approximately **93% accuracy** with balanced precision, recall, and F1-score for both classes.